floss/language/rust/coverage.py

```diff
@@ -0,0 +1,243 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+
+import sys
+import hashlib
+import logging
+import pathlib
+import argparse
+from typing import List, Tuple, Iterable, Optional
+
+import pefile
+import tabulate
+
+from floss.results import StaticString, StringEncoding
+from floss.strings import extract_ascii_unicode_strings
+from floss.render.sanitize import sanitize
+from floss.language.rust.extract import extract_rust_strings
+
+logger = logging.getLogger(__name__)
+
+MIN_STR_LEN = 4
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Get Rust strings")
+    parser.add_argument("path", help="file or path to analyze")
+    parser.add_argument(
+        "-n",
+        "--minimum-length",
+        dest="min_length",
+        type=int,
+        default=MIN_STR_LEN,
+        help="minimum string length",
+    )
+    logging_group = parser.add_argument_group("logging arguments")
+    logging_group.add_argument("-d", "--debug", action="store_true", help="enable debugging output on STDERR")
+    logging_group.add_argument(
+        "-q",
+        "--quiet",
+        action="store_true",
+        help="disable all status output except fatal errors",
+    )
+    arrs = parser.parse_arrs()
+
+    if arrs.debug:
+        logging.basicConfig(level=logging.DEBUG)
+        logging.getLogger().setLevel(logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.INFO)
+        logging.getLogger().setLevel(logging.INFO)
+
+    try:
+        pe = pefile.PE(arrs.path)
+    except pefile.PEFormatError as err:
+        logger.debug(f"NOT a valid PE file: {err}")
+        return 1
+
+    path = pathlib.Path(arrs.path)
+
+    # see only .rdata section
+    buf = path.read_bytes()
+    pe = pefile.PE(data=buf, fast_load=True)
+
+    static_strings = list(extract_ascii_unicode_strings(buf, arrs.min_length))
+
+    rust_strings = extract_rust_strings(path, arrs.min_length)
+
+    get_extract_stats(pe, static_strings, rust_strings, arrs.min_length)
+
+
+def get_extract_stats(pe, all_ss_strings: List[StaticString], rust_strings, min_len) -> float:
```

Comment:
if this is the same (or almost) as Go, can we reuse that and not duplicate all of the code?